### DISCUSSION TITLE
API: Getting an event's General Settings (Text/Images)  and Tickets customs Questions via API?

### DISCUSSION THREAD

**Discussion by fabricelaborie** [pretix/pretix#3691] [Q&A] [answered]
Hiya Team... 
How can I get more details from an event's General Settings:

1.  Texts:
- "Frontpage text" ( in each language)
- "Confirmation text"

2. Shop design:
- Header Image
- Social Media Image

Also, How Can I get the list of Custom Fields for a specific ticket/item
( I see i can get the "answers" to those questions in GET /api/v1/organizers/(organizer)/events/(event)/orders/ ) 
But if I want to POST, i need to know ( dynamically ) what QUESTIONS I should be asking ( and that's a property of the ticket ) to send across.
Thanks.

**Comment by raphaelm** [comment]
Have a look at the settings API:
https://docs.pretix.eu/en/latest/api/resources/events.html#event-settings

And the questions:
https://docs.pretix.eu/en/latest/api/resources/questions.html

**Reply by fabricelaborie** [reply]
Thanks for the quick response, Raphael.

For "questions",  it works! yeahhh . I was not expecting to see the Questions at the "Event" level but at a Ticket level, but now I see that all questions are returned , and include a list of Items to which they pertain . so that's GREAT . thanks @raphaelm ! 

But in event-settings, I still don't see the data I'm after :  I do see the locale , but nothing from Texts in each language , nor reference to 
 the images...

**Reply by raphaelm** [reply]
The four settings you are looking for are named:
- `frontpage_text`
- `confirm_texts`
- `logo_image`
- `og_image`

**Reply by fabricelaborie** [reply]
Those 4 settings are **nowhere** to be found .. ( not even the  "imprint_url": "https://pretix.eu" ) 
when I GET https://pretix.eu/api/v1/organizers/fablabtest01/events/Soundstorm2023/settings/?explain =true 
it returns the below (  I confirm that I have set the texts / logos in the web front end....  and without the explain
{
    "locales": [
        "en",
        "ar"
    ],
    "locale": "en",
    "last_order_modification_date": null,
    "show_quota_left": true,
    "max_items_per_order": 10,
    "attendee_names_asked": true,
    "attendee_names_required": false,
    "attendee_emails_asked": false,
    "attendee_emails_required": false,
    "attendee_addresses_asked": false,
    "attendee_addresses_required": false,
    "attendee_company_asked": false,
    "attendee_company_required": false,
    "ticket_download": true,
    "ticket_download_addons": false,
    "ticket_download_nonadm": true,
    "ticket_download_pending": false,
    "invoice_address_asked": true,
    "invoice_address_required": false,
    "invoice_address_vatid": false,
    "invoice_address_company_required": false,
    "invoice_address_beneficiary": false,
    "invoice_address_custom_field": {
        "en": ""
    },
    "invoice_name_required": false,
    "invoice_address_not_asked_free": false,
    "invoice_address_from_name": "",
    "invoice_address_from": "",
    "invoice_address_from_zipcode": "",
    "invoice_address_from_city": "",
    "invoice_address_from_country": "SA",
    "invoice_address_from_tax_id": "",
    "invoice_address_from_vat_id": "",
    "name_scheme": "given_family",
    "reusable_media_type_barcode": false,
    "reusable_media_type_nfc_uid": false,
    "reusable_media_type_nfc_mf0aes": false,
    "reusable_media_type_nfc_mf0aes_random_uid": false,
    "system_question_order": {},
    "pretixpos_additional_receipt_text": null,
    "_name_scheme_fields": [
        {
            "key": "given_name",
            "label": "Given name",
            "weight": 1
        },
        {
            "key": "family_name",
            "label": "Family name",
            "weight": 1
        }
    ],
    "_name_scheme_salutations": [
        {
            "key": "Ms",
            "label": "Ms"
        },
        {
            "key": "Mr",
            "label": "Mr"
        },
        {
            "key": "Mx",
            "label": "Mx"
        }
    ],
    "_name_scheme_titles": []
}

**Reply by raphaelm** [reply]
What kind of API authentication are you using?

**Reply by fabricelaborie** [reply]
Hi @raphaelm  ... I used Device Authentication .  I checked I am still authenticated as I'm able to query /organizers for example...
I saw my account got activated today, not sure if that makes a difference?

**Reply by fabricelaborie** [reply]
for good measure, since the account got activated after I created the device , i created a new device and tested again with device authentication . no luck getting frontpage_text / confirm_texts /  logo_image /  og_image :(

**Reply by raphaelm** [reply, marked as answer]
Yes, in fact it's due to the device auth. For safety and performance reasons, device auth currently has only access to a subset of the settings parameters. I think that might be missing in the documentation.

Would you mind describing your use case a little so we can think this through more thoroughly?

**Reply by fabricelaborie** [reply]
Thank you @raphaelm! Spot on, that was the issue: I changed to Token-based authentication (https://docs.pretix.eu/en/latest/api/tokenauth.html#rest-tokenauth) and it worked. I  don't need 'Device Authentication', so no need to take you through any use case... But yeah. probably a one-liner in the documentation could save someone else the same headache.

**Reply by raphaelm** [reply]
I'll add it to the docs!
